src/lib/y2partitioner/format_mount/root_subvolumes_builder.rb

```diff
@@ -0,0 +1,248 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2storage/subvol_specification"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module FormatMount
+    # Class to manage root subvolumes in expert partitioner
+    class RootSubvolumesBuilder
```

Comment:
I find this approach to singleton quite confusing because the public instance API is not really public (since there is no reasonable way to get an instance of `RootSubvolumesBuilder`). The public API of an instance is in fact the "internal" API used by the class, than then offer the really public API of the whole thing.

So, for example, `reload` or `remove_shadowed_subvolumes` are never called by the users of the class although they are public and need to remain public for the class methods to manipulate the encapsulated instance.

Too much brain twisting for my taste.